From 9f90ee5eeccd47f39c7a03dcd786b125a19c195d Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Sat, 13 Jun 2015 22:52:33 -0500
Subject: [PATCH] genmarshal: silence register storage class warnings

Using the register keyword triggers warnings on noteworthy compilers
(clang), since it's deprecated in C++ and at danger of being removed
from the language. There is no reason to use it since it isn't 1980
anymore.

https://bugzilla.gnome.org/show_bug.cgi?id=750918
---
 gobject/glib-genmarshal.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gobject/glib-genmarshal.c b/gobject/glib-genmarshal.c
index be4151a..ca78a6f 100644
--- a/gobject/glib-genmarshal.c
+++ b/gobject/glib-genmarshal.c
@@ -412,9 +412,9 @@ generate_marshal (const gchar *signame,
       g_fprintf (fout, "%s%s data2);\n", indent (ind), pad ("gpointer"));
 
       /* cfile marshal variables */
-      g_fprintf (fout, "  register GMarshalFunc_%s callback;\n", signame);
-      g_fprintf (fout, "  register GCClosure *cc = (GCClosure*) closure;\n");
-      g_fprintf (fout, "  register gpointer data1, data2;\n");
+      g_fprintf (fout, "  GMarshalFunc_%s callback;\n", signame);
+      g_fprintf (fout, "  GCClosure *cc = (GCClosure*) closure;\n");
+      g_fprintf (fout, "  gpointer data1, data2;\n");
       if (sig->rarg->setter)
 	g_fprintf (fout, "  %s v_return;\n", sig->rarg->ctype);
 
-- 
2.4.2